projects/plugins/jetpack/_inc/lib/admin-pages/class-jetpack-search-dashboard-page.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * A class that adds a search dashboard to wp-admin.
+ *
+ * @package automattic/jetpack
+ */
+
+/**
+ * Requires files needed.
+ */
+require_once JETPACK__PLUGIN_DIR . '_inc/lib/admin-pages/class.jetpack-admin-page.php';
+require_once JETPACK__PLUGIN_DIR . '_inc/lib/admin-pages/class-jetpack-redux-state-helper.php';
+
+/**
+ * Responsible for adding a search dashboard to wp-admin.
+ *
+ * @package Automattic\Jetpack\Search
+ */
+class Jetpack_Search_Dashboard_Page extends Jetpack_Admin_Page {
+	/**
+	 * Show the settings page only when Jetpack is connected or in dev mode.
+	 *
+	 * @var bool If the page should be shown.
+	 */
+	protected $dont_show_if_not_active = true;
+
+	/**
+	 * Add page specific actions given the page hook.
+	 *
+	 * @param {object} $hook The page hook.
+	 */
+	public function add_page_actions( $hook ) {}// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+
+	/**
+	 * Create a menu item for the page and returns the hook.
+	 */
+	public function get_page_hook() {
+
+		if ( ! $this->should_add_sub_menu() ) {
+			return;
+		}
+		return add_submenu_page(
+			'jetpack',
+			__( 'Search Settings', 'jetpack' ),
+			__( 'Search', 'jetpack' ),
+			'manage_options',
+			'jetpack-search',
+			array( $this, 'render' ),
+			$this->get_link_offset()
+		);
+	}
+
+	/**
+	 * Enqueue and localize page specific scripts
+	 */
+	public function page_admin_scripts() {
+		$this->load_admin_styles();
+		$this->load_admin_scripts();
+	}
+
+	/**
+	 * Override render funtion
+	 */
+	public function render() {
+		$this->page_render();
+	}
+
+	/**
+	 * Render Search setting elements
+	 */
+	public function page_render() {
+		?>
+		<div id="jp-search-dashboard" class="jp-search-dashboard">
+			<div class="hide-if-js"><?php esc_html_e( 'Your Search dashboard requires JavaScript to function properly.', 'jetpack' ); ?></div>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Test whether we should show Search menu.
+	 *
+	 * @return {boolean} Show search sub menu or not.
+	 */
+	protected function should_add_sub_menu() {
+		// TODO: temporary flag for testing. Will be removed on the last PR merge.
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		return Jetpack_Plan::supports( 'search' ) && array_key_exists( 'a8ctest', $_GET );
```

Comment:
Should we check if the site is connected / not in offline mode here?

Reply:
If the site isn't connected or is offline, I see two paths forward:

- Don't add the search dashboard submenu link. This falls back on the existing connection UX built into Jetpack.
- Add the submenu link, but use the dashboard to direct the user to the connection process. (Or to bring them out of offline mode? I'm not very familiar with offline mode.)

Context: We'd like to eventually make this the landing page when Jetpack Search becomes its own plugin via RNA.

Reply:
> Should we check if the site is connected / not in offline mode here?

This class extends `Jetpack_Admin_Page` which does the test before calling the function to add the submenu.
https://github.com/Automattic/jetpack/blob/367da565388df030e9c2ef760cfdb916b912f5ff/projects/plugins/jetpack/_inc/lib/admin-pages/class.jetpack-admin-page.php#L69

Reply:
> If the site isn't connected or is offline, I see two paths forward:
> 
>     * Don't add the search dashboard submenu link. This falls back on the existing connection UX built into Jetpack.
> 
>     * Add the submenu link, but use the dashboard to direct the user to the connection process. (Or to bring them out of offline mode? I'm not very familiar with offline mode.)
> 
> 
> Context: We'd like to eventually make this the landing page when Jetpack Search becomes its own plugin via RNA.

It currently does the first, but we could def talk about the second.

Reply:
> This class extends `Jetpack_Admin_Page` which does the test before calling the function to add the submenu.

That's perfect for now I think 👍